### PR TITLE
ESX:virt-who config CLI cases support for SCA enabled

### PR DESCRIPTION
**Pull request by yanpliu**
Relealized ESX:virt-who config CLI cases support for SCA enabled
CLI cases: PASS
```
(robottelo_vv) [root@dell-per740-68-vm-04 robottelo]# pytest ./tests/foreman/virtwho/cli/test_esx_sca.py 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.2.1, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /root/gitrepo/robottelo, configfile: pyproject.toml
plugins: xdist-3.1.0, services-2.2.1, mock-3.10.0, reportportal-5.1.3, ibutsu-2.2.4, cov-3.0.0
collected 12 items / 12 deselected / 0 selected                                                                                                                                                                   

tests/foreman/virtwho/cli/test_esx_sca.py ............                                                                                                                                                      [100%]

================================================================================================ warnings summary =================================================================================================


-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================================================== 12 passed, 12 deselected, 14 warnings in 1311.20s (0:21:51) ===========================================================================


```

### CODE REVIEW

File: tests/foreman/virtwho/cli/test_esx_sca.py [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+    @pytest.mark.tier2
+    def test_positive_rhsm_option(
+        self, module_sca_manifest_org, form_data, virtwho_config, target_sat
+    ):
+        """Verify rhsm options in the configure file"
+
+        :id: 2aad374a-c493-4e3c-91e3-60f21181fd29
+
+        :expectedresults:
+            1. rhsm_hostname, rhsm_prefix are ecpected
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/cli/test_esx_sca.py:377]:
```suggestion
            1. rhsm_hostname, rhsm_prefix are expected
```

File: tests/foreman/virtwho/cli/test_esx_sca.py [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+    ):
+        """Verify "hammer virt-who-config deploy"
+
+        :id: 04f2cef8-c88e-4a21-9d2f-c17238eea308
+
+        :expectedresults: Config can be created and deployed
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/cli/test_esx_sca.py:78]: Already defined at module level.

File: tests/foreman/virtwho/cli/test_esx_sca.py [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+        virt_who_instance = target_sat.cli.VirtWhoConfig.info({'id': virtwho_config['id']})[
+            'general-information'
+        ]['status']
+        assert virt_who_instance == 'OK'
+
+    @pytest.mark.tier2
+    def test_positive_deploy_configure_hypervisor_password_with_special_characters(
+        self, module_sca_manifest_org, form_data, target_sat
+    ):
+        """Verify " hammer virt-who-config deploy hypervisor with special characters"
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/cli/test_esx_sca.py:456]:
```suggestion
        """Verify "hammer virt-who-config deploy hypervisor with special characters"
```

File: tests/foreman/virtwho/cli/test_esx_sca.py [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+    }
+    return form
+
+
+@pytest.fixture()
+def virtwho_config(form_data, target_sat):
+    virtwho_config = target_sat.cli.VirtWhoConfig.create(form_data)['general-information']
+    yield virtwho_config
+    target_sat.cli.VirtWhoConfig.delete({'name': virtwho_config['name']})
+    assert not target_sat.cli.VirtWhoConfig.exists(search=('name', form_data['name']))
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/cli/test_esx_sca.py:62]: We can also add this  teardown in other modules where virtwho_config is defined and remove it from every test's teardown, what do you think?

File: tests/foreman/virtwho/cli/test_esx_sca.py (hunk 2 above)
Comment (yanpliu) [reply tests/foreman/virtwho/cli/test_esx_sca.py:76]: > `CaseImportance` already defined at module level.

Thanks @shubhamsg199 Have already removed the global CaseImportance

File: tests/foreman/virtwho/cli/test_esx_sca.py (hunk 4 above)
Comment (yanpliu) [reply tests/foreman/virtwho/cli/test_esx_sca.py:60]: > We can also add this teardown in other modules where virtwho_config is defined and remove it from every test's teardown, what do you think?

Yes, I really agree with you, and I also plan to do this tasks in the future in optimized the cases, also will  use fixture to parameterize some similar cases.

**Review by JacobCallahan — APPROVED** [review]
ACK. Merge pending @shubhamsg199's comments being resolved

File: tests/foreman/virtwho/cli/test_esx_sca.py (hunk 4 above)
Comment (shubhamsg199) [reply tests/foreman/virtwho/cli/test_esx_sca.py:60]: Great, Thanks!

**Review by shubhamsg199 — APPROVED** [review]
Ack

File: tests/foreman/virtwho/cli/test_esx_sca.py [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+        :BZ: 1637042, 1769680
+        """
+        data = hypervisor_json_create(hypervisors=100, guests=10)
+        url = f"{target_sat.url}/rhsm/hypervisors/{function_org.label}"
+        auth = (settings.server.admin_username, settings.server.admin_password)
+        result = requests.post(url, auth=auth, verify=False, json=data)
+        if result.status_code != 200:
+            if "foreman_tasks_sync_task_timeout" in result.text:
+                task_id = re.findall('waiting for task (.*?) to finish', result.text)[-1]
+                target_sat.wait_for_tasks(search_query=f'id = {task_id}', max_tries=10)
```
Comment (jyejare) [inline tests/foreman/virtwho/cli/test_esx_sca.py:424]: I think you missed asserting the success result code after wait_for_tasks! Please do add !

**Review by jyejare — COMMENTED** [review]
ACK pending comment !

**Comment by JacobCallahan** [comment]
@yanpliu can you check Jitendra's comment and resolve the merge conflict?

File: tests/foreman/virtwho/cli/test_nutanix.py [hunk 6]
```diff
@@ -181,7 +181,7 @@ def test_positive_prism_central_deploy_configure_by_id_script(
     ):
         """Verify "hammer virt-who-config deploy" on nutanix prism central mode
 
-        :id: 224ad753-6186-4b09-a72c-458839a8e412
+        :id: e85d7c19-ecf7-4ddd-a66b-a429e18fdc23
```
Comment (Gauravtalreja1) [inline tests/foreman/virtwho/cli/test_nutanix.py:184]: @yanpliu Could you revert this change for the test case ids? it seems irrelevant.

**Review by Gauravtalreja1 — CHANGES_REQUESTED** [review]

File: tests/foreman/virtwho/cli/test_esx_sca.py (hunk 5 above)
Comment (yanpliu) [reply tests/foreman/virtwho/cli/test_esx_sca.py:424]: > I think you missed asserting the success result code after wait_for_tasks! Please do add !

@jyejare Jentedra, 200 is the success code what we need to assert.

Comment (yanpliu) [reply tests/foreman/virtwho/cli/test_esx_sca.py:424]: > > I think you missed asserting the success result code after wait_for_tasks! Please do add !
> 
> @jyejare Jentedra, 200 is the success code what we need to assert.

          ``  else:
                assert result.status_code == 200```

File: tests/foreman/virtwho/cli/test_nutanix.py (hunk 6 above)
Comment (yanpliu) [reply tests/foreman/virtwho/cli/test_nutanix.py]: @Gauravtalreja1 Have already done.
